inginious/frontend/templates/admin/admin_users.html

```diff
@@ -0,0 +1,334 @@
+          $(this).text($(event.relatedTarget).data('username'));
+      });
+      $('.modal-dialog #username').val($(event.relatedTarget).data('username'));
+    });
+    $('#binding_modal').on('show.bs.modal', function (event) {
+        $('.modal-dialog #username').val($(event.relatedTarget).data('username'));
+        var username=$('#username').val();
+        var bindings = get_bindings(username);
+        for (const elem in bindings) {
+            var template = $('#hidden-template').clone();
+            template = $(template).children("div.card.mb-3").first().attr('id',elem+'-template').parent().html();
+            $("#binding_content").append(template);
+            $("#"+elem+"-template .binding_revoke").data("username",username);
+            $("#"+elem+"-template .binding_revoke").data("binding",elem);
+            $("#"+elem+"-template .binding_identifier").text(bindings[elem][0]);
+            $("#"+elem+"-template .binding_method").text(elem);
+        }
+        $('.binding_revoke').bind("click", revoke_binding);
+    });
+    $('#binding_modal').on('hide.bs.modal', function (event) {
+        $("#binding_content").html('');
+    });
+    $(".delete_user").each(function(index) {
+        $(this).tooltip({"placement": "bottom"});
+    });
+    $(".activate_user").each(function(index) {
+        $(this).tooltip({"placement": "bottom"});
+    });
+    $("button:submit").on("click",function(){
+        var username=$('#username').val();
+        var action = $(this).data("action");
+        var realname = "";
+        var email = "";
+        var password = "";
+        if (action == "add_user"){
+            username = $('input[name="usrname"]').val();
+            realname = $('input[name="realname"]').val();
+            email = $('input[name="email"]').val();
+            password = $('input[name="password"]').val();
+        }
+        $.ajax({
+          type: "POST",
+          url: "{{get_homepath()}}/administrator/user_action",
+          data: {
+            "username":username,
+            "realname":realname,
+            "email": email,
+            "password":password,
+            "action":action
+          },
+        }).done(function(feedback) {
+            if (feedback == ""){
+                window.location.href = "{{get_homepath()}}/administrator/users";
+            }
+            $('#feedback').text(feedback);
+
+        });
+    });
+    function revoke_binding(event){
+        var username = $(this).data("username");
+        var binding_id = $(this).data("binding");
+        var action ="revoke_binding";
+        $.ajax({
+          type: "POST",
+          url: "{{get_homepath()}}/administrator/user_action",
+          data: {
+            "username":username,
+            "action":action,
+            "binding_id": binding_id
+          },
+        }).done(function() {
+          window.location.href = "{{get_homepath()}}/administrator/users";
```

Comment:
Is this necessary ? Closing the modal isn't enough ? You will loose the current displayed page information here.